### DISCUSSION TITLE
Tailwind with yui CSS compressor enabled

### DISCUSSION THREAD

**Discussion by the-pulli** [rails/tailwindcss-rails#243] [Q&A]
Are there any strategies how to use Tailwind with yui as a CSS compressor?

If I enable yui, the already minified tailwind.css from (app/assets/builds) gets again minified (and messed up) by yui and copied as an asset to public/assets.

Can I configure Rails to exclude the tailwind.css from being minified again?

I reproduced the behaviour with yui directly in my console with same results as in my Rails app.

Attached you find the example files from my app:
[tailwind-without-yui.txt](https://github.com/rails/tailwindcss-rails/files/10569209/tailwind-without-yui.txt)
[tailwind-yui.txt](https://github.com/rails/tailwindcss-rails/files/10569224/tailwind-yui.txt)

Any help is appreciated, thanks 🙃

**Comment by flavorjones** [comment]
@the-pulli Since v2.0.13, you can generate unminified `tailwind.css` by passing the "debug" argument to the rake task:

```
bin/rails tailwindcss:build[debug]
# or
bin/rails tailwindcss:watch[debug]
```

Does that meet your needs? You may need to update your `Procfile.dev` if that's the workflow you're using.

**Reply by flavorjones** [reply]
This is documented in the README here: https://github.com/rails/tailwindcss-rails#debugging-with-unminified-assets

**Comment by the-pulli** [comment]
@flavorjones, thanks! That goes in the right direction. But in production I call `bin/rails assets:precompile`, which calls then `bin/rails tailwindcss:build`, if I understood it right. How do I pass the `debug` there? Or am I missing something here?

**Reply by flavorjones** [reply]
Ah, you're right, when it's invoked as a dependency the `debug` argument can't be injected easily.

I'm going to create an issue from this discussion, I think I can get to that in the next few days.

**Reply by flavorjones** [reply]
See #244

**Reply by the-pulli** [reply]
Awesome! Thanks so much for your help & support 👍

**Comment by dhh** [comment]
Why do you want to compress TW with yui?

**Reply by the-pulli** [reply]
If I don't set a CSS compressor my normal CSS won't get compressed. If I set yui as compressor, it compresses TW again and messes up the file.
Or am I doing something wrong here?

**Comment by the-pulli** [comment]
@flavorjones So you closed your PR https://github.com/rails/tailwindcss-rails/pull/248. Will there be an implementation of that feature at another place in the Rails codebase?

**Reply by flavorjones** [reply]
My understanding off DHH's guidance on that PR was that (if things break otherwise) you should try to configure your minifier to avoid minifying the tailwindcss file.

**Reply by the-pulli** [reply]
Ok, so was my understanding too. But I'm not sure how to tell Rails to exclude the TW css from minification.

**Reply by flavorjones** [reply]
@the-pulli Can I ask that you create an issue in rails/rails to start a conversation with the rails core team about it? It may be worth explaining the setup in detail there (my understanding is that you're using `Rails.application.config.assets.css_compressor = :yui` and seeing broken styling, that would be useful to provide there so context is all in one place).

**Reply by the-pulli** [reply]
Done. See https://github.com/rails/rails/issues/47553.
